src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,89 @@
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+
+import static seedu.addressbook.ui.TextUi.DISPLAYED_INDEX_OFFSET;
+
+/**
+ * Edits a person identified using it's last displayed index from the address book.
+ */
+public class EditCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits a person identified using it's last displayed index from the address book.\n"
+            + "Contact details can be marked private by prepending 'p' to the prefix.\n"
+            + "Parameters: INDEX  NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Example: " + COMMAND_WORD
+            + " 1 John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Old entry: %1$s\n";
+    public static final String MESSAGE_SUCCESS = "New entry: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+
+    private final Person toEdit;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(String name,
+                      String phone, boolean isPhonePrivate,
+                      String email, boolean isEmailPrivate,
+                      String address, boolean isAddressPrivate,
+                      Set<String> tags,
+                      int targetVisibleIndex) throws IllegalValueException {
+        super(targetVisibleIndex);
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toEdit = new Person(
+                new Name(name),
+                new Phone(phone, isPhonePrivate),
+                new Email(email, isEmailPrivate),
+                new Address(address, isAddressPrivate),
+                tagSet
+        );
+    }
+
+    public EditCommand(Person toEdit, int targetVisibleIndex) {
+        super(targetVisibleIndex);
+        this.toEdit = toEdit;
+    }
+
+    public ReadOnlyPerson getPerson() {
+        return toEdit;
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.editPerson(toEdit, getTargetIndex() - DISPLAYED_INDEX_OFFSET);
+
+            String result = String.format(MESSAGE_EDIT_PERSON_SUCCESS, target) +
+                    String.format(MESSAGE_SUCCESS, toEdit);
```

Comment:
for line wrap, according to our coding standard guidelines, the `+` operator should be on the next line, otherwise this line of code will just look like a valid code that is misindented.